Lib/test/test_signal.py

```diff
@@ -423,6 +423,74 @@ def handler(signum, frame):
         """.format(action=action)
         assert_python_ok('-c', code)
 
+    @unittest.skipIf(_testcapi is None, 'need _testcapi')
+    def test_warn_on_full_buffer(self):
```

Comment:
Please try to split this test into 3 tests, and handle stderr in the parent process. Running 3 tests in the same process may have side effects between tests.

Reply:
I'd rather not? Part of what I want to test is that there are no side-effects between tests :-). (As there could be e.g. if `warn_on_full_buffer` values leaked between different calls to `set_wakeup_fd`.) And the handling of stderr within the child process is copied straight from the other wakeup fd warning tests in the same file.

Reply:
Ok.